0000-access-keys.md

```diff
@@ -0,0 +1,313 @@
+        // limited the spending amount to some reasonable, but large enough number.
+        // NOTE: It's needs to be multiplied to decimals, e.g. 10^-18 
+        allowance: 1_000_000_000,
+        
+        // This access key restricts access to `chess.app` contract.
+        receiver_id: "chess.app",
+
+        // Any method name on the `chess.app` contract can be called.  
+        method_name: None,
+    }),
+}
+```
+
+#### AccessKey issued by a dApp
+
+This is an example where the dApp wants to pay for the user, or it doesn't want to go through the user's sign-in flow.
+For whatever reason the dApp decided to issue an access key directly for their account.
+
+For this to work there should be one account with funds (that dApp controls on the backend) which creates access keys for the users.
+The difference from the example above is there is only one account (the same for all users) that creates multiple access keys (one per user) towards one other contract (app's contract).
+To differentiate users the contract has to use the public key of the access key instead of sender's account ID.
+
+If the access key wants to support user's identity from the account ID. The contract can provide a public method that links user's account ID with a given public key.
+Once this is done, a user can request a new access key with the linked public key (sponsored by the app), but it is linked to the user's account ID.
+
+There are some caveats with this approach:
+- The dApp is required to have a backend and to have some sybil resistance for users. It's needed to prevent abuse by bots.
+- Writing the contract is slightly more complicated, since the contract now needs to handle mapping of the public keys to the account IDs.
+
+An access key example for chess app paid by the chess app from `chess.funds` account:
+```rust
+/// chess.funds,2bc2b3b
+AccessKey {
+    nonce: 0,
+    
+    permission: AccessKeyPermission::FunctionCall(FunctionCallPermission {
+        // Since the access key is given to the user, the developer wants to limit the
+        // the spending amount to some conservative number, since a user might try to drain it.
+        allowance: 5_000_000,
+        
+        // This access key restricts access to `chess.app` contract.
+        receiver_id: "chess.app",
+
+        // Any method name on the `chess.app` contract can be called (but some methods might just ignore this key).
+        method_name: None,
+    }),
+}
+```
+
+#### AccessKey through a proxy
+
+This examples demonstrates how to have more granular control on top of built-in access key restrictions.
+
+Let's say a user wants to:
+- limit the number of calls the access key can make per minute
+- support multiple contracts with the same access key
+- select which methods name can be called and which can't
+- transfer funds from the account up to a certain limit 
+- stake from the account, but prevent withdrawing funds
```

Comment:
Add example with social network to explain how we can use proxy to attribute account id with a public key.

Reply:
We decided to avoid this proxy example and rely on `authorize` public key to link an identity with the given public key